File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    # dp
    # 첫번째 칸으로 가는 경우 1
    # 두번째 칸 2
    # 세번째 칸부터는 dp[i] = dp[i-1] + dp[i-2]
    dp = [0]*n
    if n < 3:
        return n
    dp[0] = 1
    dp[1] = 2
    for i in range(2,n):
        dp[i] = (dp[i-1] + dp[i-2])%1234567
    return dp[n-1]

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        /*
        dp문제
        dp[0] = 1
        dp[1] = 1
        dp[2] = dp[1] + dp[0] 2
        3 -> 3
        4 -> 5
        ...
        (n>1)dp[n] = dp[n-1] + dp[n-2]
        */
        long[] dp = new long[n+1];
        dp[0] = 1;
        dp[1] = 1;
        for(int i=2; i<=n; i++){
            dp[i] = (dp[i-1] + dp[i-2])%1234567;
        }
        return dp[n];
    }
}